src/Request/Data/AmountData.php

```diff
@@ -30,10 +32,11 @@ public function setAmount(string $amount): AmountData
     }
 
     /**
-     * @param string $code
+     * @param CurrencyEnum $code
+     *
      * @return $this
      */
-    public function setCode(string $code): AmountData
+    public function setCode(CurrencyEnum $code): AmountData
```

Comment:
To avoid breaking change, deprecate the method, create a new one that is setCodeEnum